main/src/ca/uwaterloo/flix/api/lsp/provider/completion/KeywordDeclCompleter.scala

```diff
@@ -0,0 +1,37 @@
+
+/*
+ * Copyright 2023 Magnus Madsen, Alexander Dybdahl Troelsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.Index
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+object KeywordDeclCompleter extends Completer {
+
+  def getCompletions(context: CompletionContext)(implicit flix: Flix, index: Index, root: TypedAst.Root): Iterable[Completion] = 
+    List(      
+      "def",
+      "eff",
+      "enum",
+      "instance",
+      "mod",
+      "pub",
+      "sealed",
+      "trait",
+      "with",
```

Comment:
You cannot write with by itself, right? Its not a declaration.

Reply:
I see. I misunderstood. I thought the keywords which should be included were any that would be relevant within a declaration, not just those that were declaration keywords. Like you write in another comment, I think id benefit from looking more into how syntactic contexts are created and used

Reply:
Indeed. Just remember the overall point: We should only suggest things that are meaningful.